daemon/daemon.go

```diff
@@ -579,14 +579,21 @@ func (d *Daemon) Stop(sigCh chan<- os.Signal) error {
 
 	err := d.tomb.Wait()
 	if err != nil {
-		// do not stop the shutdown even if the tomb errors
-		// because we already scheduled a slow shutdown and
-		// exiting here will just restart snapd (via systemd)
-		// which will lead to confusing results.
-		if restartSystem {
-			logger.Noticef("WARNING: cannot stop daemon: %v", err)
+		if err == context.DeadlineExceeded {
+			logger.Noticef("WARNING: cannot gracefully shut down the snapd API: %v", err)
```

Comment:
we have actually closed the listeners already, so this log is a bit misleading:

"WARNING: cannot gracefully  shut down in-flight snapd API activity within %v", shutdownTimeout

the error is fixed here afaiu